packages/gatsby-source-contentful/src/__tests__/gatsby-node.js

```diff
@@ -579,44 +579,89 @@ describe(`gatsby-node`, () => {
     expect(actions.deleteNode).toHaveBeenCalledTimes(0)
     expect(actions.touchNode).toHaveBeenCalledTimes(0)
     expect(reporter.info.mock.calls).toMatchInlineSnapshot(`
-      Array [
-        Array [
+      [
+        [
           "Contentful: 4 new entries",
         ],
-        Array [
+        [
           "Contentful: 0 updated entries",
         ],
-        Array [
+        [
           "Contentful: 0 deleted entries",
         ],
-        Array [
+        [
           "Contentful: 0 cached entries",
         ],
-        Array [
+        [
           "Contentful: 4 new assets",
         ],
-        Array [
+        [
           "Contentful: 0 updated assets",
         ],
-        Array [
+        [
           "Contentful: 0 cached assets",
         ],
-        Array [
+        [
           "Contentful: 0 deleted assets",
         ],
-        Array [
+        [
           "Creating 1 Contentful Person nodes",
         ],
-        Array [
+        [
           "Creating 3 Contentful Blog Post nodes",
         ],
-        Array [
+        [
           "Creating 4 Contentful asset nodes",
         ],
       ]
     `)
   })
 
+  it(`should create nodes with custom prefix`, async () => {
+    // @ts-ignore
+    fetchContent.mockImplementationOnce(startersBlogFixture.initialSync)
+    schema.buildObjectType.mockClear()
+    // @ts-ignore
+    await simulateGatsbyBuild({
+      typePrefix: `CustomPrefix`,
+    })
+
+    expect(schema.buildObjectType).toHaveBeenCalledTimes(3)
+    expect(schema.buildObjectType).toHaveBeenCalledWith(
+      expect.objectContaining({
+        name: `CustomPrefixPerson`,
+      })
+    )
+    expect(schema.buildObjectType).toHaveBeenCalledWith(
+      expect.objectContaining({
+        name: `CustomPrefixBlogPost`,
+      })
+    )
+    expect(schema.buildObjectType).toHaveBeenCalledWith(
+      expect.objectContaining({
+        name: `CustomPrefixAsset`,
+      })
+    )
+
+    expect(schema.buildObjectType).not.toHaveBeenCalledWith(
+      expect.objectContaining({
+        name: `ContentfulPerson`,
+      })
+    )
+
+    expect(schema.buildObjectType).not.toHaveBeenCalledWith(
+      expect.objectContaining({
+        name: `ContentfulBlogPost`,
+      })
+    )
+
+    expect(schema.buildObjectType).not.toHaveBeenCalledWith(
+      expect.objectContaining({
+        name: `ContentfulAsset`,
+      })
+    )
+  })
```

Comment:
Checking both that it created the types _with_ the prefix, but also that it doesn't create types with the default